File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        if (nums.length == 0) {
            return false;
        }

        int maxReach = 0;  // The farthest point that can be reached
        for (int i = 0; i < nums.length; i++) {
            if (i > maxReach) {
                return false;  // If current index is beyond the farthest point we can reach
            }
            maxReach = Math.max(maxReach, i + nums[i]);  // Update the farthest point
            if (maxReach >= nums.length - 1) {
                return true;  // If we can reach or go beyond the last index
            }
        }

        return false;  // If the loop ends without reaching the last index
    }
}

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) == 0 or len(nums)==1:
            return True
        
        
        maxreach = 0
        
        for i in range(len(nums)-1):
            if i> maxreach:
                return False
            
            maxreach = max(maxreach, i+nums[i])
            if maxreach>= len(nums)-1:
                return True
            
        return False
        
        
        